tooling/metamodel-generator/src/main/java/org/hibernate/processor/validation/ProcessorSessionFactory.java

```diff
@@ -369,7 +369,8 @@ public EntityPersister(String entityName, TypeElement type, ProcessorSessionFact
 		@Override
 		boolean isSubclassPersister(MockEntityPersister entityPersister) {
 			EntityPersister persister = (EntityPersister) entityPersister;
-			return typeUtil.isSubtype( persister.type.asType(), type.asType() );
+			return typeUtil.isSubtype( persister.type.asType(), type.asType() )
+					&& !typeUtil.isSameType( persister.type.asType(), type.asType() );
```

Comment:
Wait, I don't think this change is quite correct.

I believe that `isSubclassEntityName()` is supposed to return `true` when passed the name of the current entity.

i.e. we consider `T` to be a subtype of `T`.